include/tester/db/Database.hpp

```diff
@@ -1,10 +1,203 @@
+    //         }   
+            
+    //         result << bsoncxx::builder::stream::close_document;
+    //         // std::cout << bsoncxx::to_json(result.view());
+
+    //         // results.emplace_back(std::move(doc));
+
+    //     }
+
+    //     return result;
+    // }
+    bsoncxx::document::value getAllHeaders(const std::string& collection_name)
+    {
+        mongocxx::collection collection = _database[collection_name];
+        mongocxx::cursor cursor = collection.find({});
+
+        bsoncxx::builder::stream::document result{};
+        bsoncxx::builder::stream::array headers{};
+
+        for (auto&& doc : cursor) {
+            bsoncxx::builder::stream::document header_builder{};
+            header_builder << "_id" << doc["_id"].get_oid().value.to_string();
+
+            if(doc["creation-time"])
+            {
+                header_builder << "creation-time" << doc["creation-time"].get_string().value.to_string();
+            }
+
+            headers << header_builder;
+        }
+
+        result << "headers" << headers;
+
+        return result.extract();
+    }
+
+    std::vector<bsoncxx::document::value> get_all_documents(const std::string& collection_name)
+    {
+        mongocxx::collection collection = _database[collection_name];
+        mongocxx::cursor cursor = collection.find({});
+
+        std::vector<bsoncxx::document::value> results;
+        for (auto&& doc : cursor) {
+            results.emplace_back(std::move(doc));
+        }
+
+        return results;
+    }
+
+    bsoncxx::stdx::optional<std::string> get_field(const std::string& collection_name, const std::string& id, const std::string& field_name)
+    {
+        mongocxx::collection collection = _database[collection_name];
+        bsoncxx::stdx::optional<bsoncxx::document::value> result = collection.find_one(bsoncxx::builder::stream::document{} << "_id" << bsoncxx::oid{id} << bsoncxx::builder::stream::finalize);
+
+        if (result) {
+            bsoncxx::document::view view = (*result).view();
+            if (view[field_name]) {
+                std::cout << std::string{ view[field_name].get_string().value.to_string() } << std::endl;
+                return std::string{ view[field_name].get_string().value.to_string() };
+            }
+        }
+
+        return bsoncxx::stdx::nullopt;
+    }
+
+
+
+    bsoncxx::stdx::optional<bsoncxx::document::value> getLog(const std::string& collection_name, const std::string& id)
+    {
+        mongocxx::collection collection = _database[collection_name];
+        bsoncxx::stdx::optional<bsoncxx::document::value> result = collection.find_one(bsoncxx::builder::stream::document{} 
+            << "_id" << bsoncxx::oid{id} << bsoncxx::builder::stream::finalize);
+
+        return result;
+    }
+
+
+    bsoncxx::stdx::optional<bsoncxx::document::value> get(const std::string &collaction_name,  const std::string &id, const std::string &time)
```

Comment:
`collectionName` - by value, and then `std::move`
`id` - same thing
`time` - same thing

Reply:
done